Hardcore.lua

```diff
@@ -152,12 +154,15 @@ function Hardcore:PLAYER_LOGIN()
 	self:RegisterEvent("UNIT_SPELLCAST_STOP")
 	self:RegisterEvent("UNIT_SPELLCAST_SUCCEEDED")
 
-	if ( Hardcore_Character.time_tracked == nil ) then
-		Hardcore_Character.time_tracked = 0
-	end
-
-	if Hardcore_Character.deaths == nil then
-		Hardcore_Character.deaths = 0
+	-- different guid + lvl1 means new character with the same name
+	if Hardcore_Character.guid ~= PLAYER_GUID then
```

Comment:
Removed the check for PLAYER_LEVEL == 1 from this condition.
This should now work for every case.

First character with name:
Hardcore_Character.guid is nil, PLAYER_GUID is the guid of the character.

New character with same name (old character was deleted):
Hardcore_Character.guid has a set value that is different from PLAYER_GUID

Character advances past level 1 before turning on the addon:
There no longer is a check for level, so this is not an issue.